File: tests/fixtures/php/reorder.php
<?php
// SHOULD flag: expensive before cheap
if ($user->isAdmin() && $isEnabled) {
    echo "yes";
}

// SHOULD flag: method call before variable
$result = ($obj->getResult() && $flag) ? 1 : 0;

// SHOULD NOT flag: already in correct order
if ($isEnabled && $user->isAdmin()) {
    echo "yes";
}

// SHOULD NOT flag: already in correct order (variable then property)
if ($active && $obj->prop) {
    echo "yes";
}

File: tests/fixtures/go/reorder.go
package main

import "strings"

func example(enabled bool, name string) {
	// SHOULD flag: expensive call before cheap variable
	if strings.Contains(name, "admin") && enabled {
		println("yes")
	}

	// SHOULD NOT flag: already correct
	if enabled && strings.Contains(name, "admin") {
		println("yes")
	}
}
